ui/components/app/gas-timing/gas-timing.component.js

```diff
@@ -1,12 +1,66 @@
-import React from 'react';
-import PropTypes from 'prop-types';
+import React, { useContext } from 'react';
 import classNames from 'classnames';
 
+import { useGasFeeEstimates } from '../../../hooks/useGasFeeEstimates';
+import { useGasFeeInputs } from '../../../hooks/useGasFeeInputs';
+import { I18nContext } from '../../../contexts/i18n';
+
 import Typography from '../../ui/typography/typography';
 import { TYPOGRAPHY } from '../../../helpers/constants/design-system';
-import InfoTooltip from '../../ui/info-tooltip';
 
-export default function GasTiming({ text, tooltipText, attitude }) {
+// Once we reach this second threshold, we switch to minutes as a unit
+const SECOND_CUTOFF = 90;
+
+export default function GasTiming() {
+  const { gasFeeEstimates, isGasEstimatesLoading } = useGasFeeEstimates();
+  const { maxPriorityFeePerGas } = useGasFeeInputs();
+
+  const t = useContext(I18nContext);
+
+  // Shows "seconds" as unit of time if under SECOND_CUTOFF, otherwise "minutes"
+  const toHumanReadableTime = (milliseconds = 1) => {
+    const seconds = Math.ceil(milliseconds / 1000);
+    if (seconds <= SECOND_CUTOFF) {
+      return t('gasTimingSeconds', [seconds]);
+    }
+    return t('gasTimingMinutes', [Math.ceil(seconds / 60)]);
+  };
+
+  // Don't show anything if we don't have enough information
+  if (isGasEstimatesLoading || maxPriorityFeePerGas < 1) {
+    return null;
+  }
+
+  const { low, medium, high } = gasFeeEstimates;
+
+  let text = '';
+  let attitude = '';
+
+  // Anything medium or faster is positive
+  if (
+    Number(maxPriorityFeePerGas) >= Number(medium.suggestedMaxPriorityFeePerGas)
```

Comment:
I'm seeing these properties as strings in a few places -- can we convert to numbers in `useGasFeeEstimates` and `useGasFeeInputs`, or is there reason not to?

Reply:
@brad-decker ^

Reply:
In some places `maxPriorityFeePerGas` is in decimal GWEI and in other places it's in hex WEI. Do we want to use some postfix to make it clearer, e.g. `maxPriorityFeePerGasHexWEI` or `maxPriorityFeePerGasDecGWEI`?